subsys/bluetooth/host/gatt.c

```diff
@@ -3145,7 +3149,15 @@ bool bt_gatt_is_subscribed(struct bt_conn *conn,
 		__ASSERT(attr, "No more attributes\n");
 	}
 
-	/* Check if the attribute is the CCC Descriptor */
+	/* Find the CCC Descriptor */
+	while (bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CCC) &&
+	       bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CHRC)) {
```

Comment:
```suggestion
	while (bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CCC) &&
	       /* But stop if we leave the current characteristic definition. */
	       bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CHRC)) &&
	       bt_uuid_cmp(attr->uuid, BT_UUID_GATT_PRIMARY)) &&
	       bt_uuid_cmp(attr->uuid, BT_UUID_GATT_SECONDARY)) {
```

Reply:
It would be great if you add test cases for a missing CCC with all the different ways to end a characteristic definition.

Reply:
we can leave that to a future unit-test I think.